tutorials/arvo/cli-tutorial.md

```diff
@@ -0,0 +1,463 @@
+      {$pro sole-prompt}                                ::  set prompt
+      {$sag p/path q/*}                                 ::  save to jamfile
+      {$sav p/path q/@}                                 ::  save to file
+      {$tab p/(list {=cord =tank})}                     ::  tab-complete list
+      {$tan p/(list tank)}                              ::  classic tank
+  ::  {$taq p/tanq}                                     ::  modern tank
+      {$txt p/tape}                                     ::  text line
+      {$url p/@t}                                       ::  activate url
+  ==
+```
+
+## `%shoe` app walkthrough {#shoe-app-walkthrough}
+
+Here we explore the capabilities of the `%shoe` example app and then go through
+the code, explaining what each line does. 
+
+### Playing with `%shoe`
+
+First let's test the functionality of `%shoe` so we know what we're getting into.
+
+Start two fake ships, one named `~zod` and the other can have any name - we will
+go with `~nus`. Fake ships run locally are able to see each other, and our
+intention is to connect their `%shoe` apps.
+
+On each fake ship start `%shoe` by entering `|start %shoe` into dojo. You may
+then switch to `%shoe` by pressing `Ctrl-X`, possibly multiple times, which will
+change the prompt to `~zod:shoe>` and `~nus:shoe>`. Enter `demo` from `~zod` and press Enter:
+```
+~zod ran the command
+~zod:shoe> 
+```
+`~zod ran the command` should be displayed in bold green text, signifying that
+the command originated locally.
+
+Now we will connect the sessions. Switch `~zod` back to dojo and enter `|link
+~nus %shoe`. If this succeeds you will see the following.
+```
+>=
+; ~nus is your neighbor
+[linked to [p=~nus q=%shoe]]
+```
+Now `~zod` will have two `%shoe` sessions running - one local one on `~zod` and
+one remote one on `~nus`, which you can access by pressing `Ctrl-X` until you see
+`~nus:shoe>` from `~zod`'s console. On the other hand, you should not see
+`~zod:shoe>` on `~nus`'s side, since you have not connected `~nus` to `~zod`'s
+`%shoe` app. When you enter `demo` from `~nus:shoe>` on
+`~zod`'s console you will again see `~zod ran the command`, but this time it
+should be in the ordinary font used by the console, signifying that the command
+is originating from a remote session. Contrast this with entering `demo` from
+`~nus:shoe>` in `~nus`'s console, which will display `~nus ran the command` in
+bold green text.
+
+Now try to link to `~zod`'s `%shoe` session from `~nus` by switching to the dojo
+on `~nus` and entering `|link ~zod %shoe`. You should see
+```
+>=
+[unlinked from [p=~zod q=%shoe]]
+```
+and if you press `Ctrl-X` you will not get a `~zod:shoe>` prompt. This is
+because the example app is set up to always allow `~zod` to connect (as well as
+subject moons if the ship happens to be a planet) but not `~nus`, so this
+message means that `~nus` failed to connect to `~zod`'s `%shoe` session.
```

Comment:
As far I know we don't document remote linking anywhere (which is why I had to ask you how to do it), so I definitely find it important to include it here. I will try to trim down the text, though.

Reply:
I didn't spend too much thinking about it, but I couldn't think of a nice way to remove more than a couple sentences since its mostly focused on the remote linking aspect and the asymmetry of the `+can-connect` arm. Perhaps remote linking could be covered in a separate tutorial that could be referenced, but it would need an example app, and `shoe` shares the bottom with other possible simplest example apps that allow remote linking, so I don't think that would be a gain. Even telling the user to run `|start` seems important here, since the only other place in the docs that command is ever mentioned is in the Egg Timer tutorial in Hoon School, and I can think of a few good reasons to want to have such an important command be mentioned more than a single time. I remember I had to ask someone (maybe you?) to remind me how to start Gall apps when I started working on this tutorial.

Reply:
> Even telling the user to run |start seems important here, since the only other place in the docs that command is ever mentioned is in the Egg Timer tutorial in Hoon School

Considering the egg timer isn't trivially discoverable as "hoon apps 101" (imo) this is a docs smell. (;